cns/service/main.go

```diff
@@ -1607,3 +1623,15 @@ func PopulateCNSEndpointState(endpointStateStore store.KeyValueStore) error {
 	}
 	return nil
 }
+
+// GetPrimaryNICMacAddress fetches the MAC address of the primary NIC on the node.
+func GetPrimaryNICMacAddress(ctx context.Context) (string, error) {
```

Comment:
@kmurudi 1. When you say "Primary NIC", does it refer to "infra NIC"? 2. Is this the correct approach to obtain the CNS's mac address?

Reply:
1. yes 
2. can we use any of the existing functions to get the primary nic (infra nic) from the wireserver client instead of using mtpnc? mtpnc will set the secondary nics' macaddress. 
refer getPrimaryHostInterface in https://github.com/Azure/azure-container-networking/blob/master/cns/restserver/util.go

not sure if `restserver` package can be imported here though, might create a cyclic package dependency if `main` is called anywhere in `restserver` package